app/code/Magento/Checkout/view/frontend/web/js/view/shipping.js

```diff
@@ -121,9 +121,7 @@ define([
                     );
                 }
                 checkoutProvider.on('shippingAddress', function (shippingAddrsData) {
-                    if (shippingAddrsData.street && !_.isEmpty(shippingAddrsData.street[0])) {
-                        checkoutData.setShippingAddressFromData(shippingAddrsData);
-                    }
+                    checkoutData.setShippingAddressFromData(shippingAddrsData);
```

Comment:
As I see result looks like revert changes for "MC-38048: Incorrect default country displayed on shipping page when store view is changed in cart. Part 2" https://github.com/magento/magento2/commit/d59ad58d924f50350908ae504bbff46ebdef59d6
As result we revert issue with store view change. Could you please recheck it.
Special thanks for mftf.